### ISSUE TITLE
Feature Request: Ability to configure `auto` file types

### ISSUE THREAD

**Issue by mhassan1** [tj/n#639]
# Problem

## Short Version

It would be useful to be able to configure which file types are crawled by `auto`

## Long Version

Support for `.nvmrc` files in the `auto` feature is awesome! Our developers use `nvm` for managing node versions in different projects, and we are using `n` for managing node versions in our CI/CD. It's cool to be able to use both tools with only one configuration file.

However, the fact that a `package.json` file (especially one without a `engines.node` setting) interrupts crawling for `.nvmrc` files makes this feature less useful in a monorepo setting. Here is one use case: https://github.com/tj/n/issues/636#issuecomment-733433947

Here is a list of cases where crawling all the file types is undesirable:
* A `package.json` file with no `engines.node` interrupts crawling for other file types
* A `package.json` file with a very permissive `engines.node` setting (e.g. `"*"`) interrupts crawling for other file types (since `engines.node` may serve a different purpose than `.nvmrc` in a developer's home directory, for example)
* A `package.json` file is found but there is no `node` or `npx` binary available yet (currently, this is an error, which penalizes the user for using `auto` even if they aren't using `engines.node`)

Some potential solutions:
* Add additional presets for single file types, e.g. `n .nvmrc` or `n auto-.nvmrc` or `n auto[.nvmrc]`
* Add support for specifying a list of file types, e.g. `n auto[.nvmrc,package.json]`
* Add a flag to skip `package.json`, similar to https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-ignore-engines (this is a stretch)

## Configuration Details

```bash
$ n --version
6.7.0

$ command -v node
/Users/me/.n/bin/node

$ node -p process.platform
darwin
```

**Comment by shadowspawn** [comment]
Are all your use cases solved by removing the scanning of `package.json`?

I almost did it separately as `n engine` originally, but went for `auto` as a single solution. 

I am interested in whether the people who want and use the `package.json` support mix it with other version sources (like `.nvmrc`), or whether having separate commands would be fine and more flexible. I'll open another issue to poll for feedback.

**Comment by shadowspawn** [comment]
Of the three potential solutions suggested, I currently prefer a flag to ignore engine/package. Scanning `package.json` is the outlier in terms of function and requirements.

**Comment by shadowspawn** [comment]
Another approach, make `package.json` a file of last resort. Only look for `package.json` after searching up hierarchy for the other (better) version control files. See #644

**Comment by shadowspawn** [comment]
Modified behaviour for `auto` released in v7.0.0
